trustfall_core/test_data/README.md

```diff
@@ -67,3 +67,26 @@ or produce a specified error ([`schema_errors`](./tests/schema_errors/) director
 The naming scheme for schema files and their corresponding expected errors is
 similar to the one for queries: schemas are named like `X.graphql` while their
 errors are named `X.schema-error.ron`.
+
+## Recreating the test files
+
+Under [`scripts`](../../scripts) there are bash scripts to create the `ron` files.
+In general, the scripts named `recreate_*` work as a "pipeline" where they take the
+output of a previous script and generate a `ron` file. For example
+`recreate_all_parsed_graphql.sh` takes all `*.graphql.ron` files under
+`trustfall_core/test_data/tests/valid_queries`, `trustfall_core/test_data/tests/execution_errors`,
+and `trustfall_core/test_data/tests/frontend_errors` and generates `*.graphql-parsed.ron` on
+those folders. While `recreate_all_ir.sh` takes all `*.graphql-parsed.ron`and generates
+the `*.ir.ron`.
+
+They are useful when you slightly modified a `*.graphql.ron` and want to update the other
+files based on that.
+
+The scripts named `create_*` receive as input a list of `graphql.ron` files and generate 
+all needed files for a given test case. For example, `create_frontend_error_data.sh` will
+create `*.graphql-parsed.ron` and `*.frontend-error.ron` files.
+
+If you want to create a new valid query test, you can run:
+`./create_valid_query_data.sh ../trustfall_core/test_data/tests/valid_queries/my_test.graphql.ron`.
+
+Scripts must be run inside the `scripts` folder.
```

Comment:
Is this true? I thought I had made them be location-independent. Could you double-check?

If any of them are not location-independent, that's also worth fixing, perhaps in the next PR.

Reply:
Yes, not sure if there are more things to change, but the `get_schema_by_name` hardcodes the path of the schema to be `../trustfall_core/test_data/schemas/{schema_name}.graphql`, which fails if you run the script from the repo root:

```
λ ~/programming/trustfall/ era/projection* ./scripts/create_valid_query_data.sh trustfall_core/test_data/tests/valid_queries/elias_test.graphql.ron
> Starting on file trustfall_core/test_data/tests/valid_queries/elias_test.graphql.ron
   Compiling trustfall_testbin v0.1.0 (/home/era/programming/trustfall/trustfall_testbin)
    Finished dev [unoptimized + debuginfo] target(s) in 2.52s
     Running `target/debug/trustfall_testbin parse trustfall_core/test_data/tests/valid_queries/elias_test.graphql.ron`
    Finished dev [unoptimized + debuginfo] target(s) in 0.04s
     Running `target/debug/trustfall_testbin frontend trustfall_core/test_data/tests/valid_queries/elias_test.graphql-parsed.ron`
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: failed to read schema from ../trustfall_core/test_data/schemas/numbers.graphql

Caused by:
    No such file or directory (os error 2)', trustfall_testbin/src/main.rs:46:10
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
But it just works if I go to the scripts folder (well, and any other folder that is just 1 level depth in the repo).

I can change in this or the next PR, is up to you :)

Reply:
Nice catch! I thought `cargo run` imposed a bit more control over the pwd where the program runs. Let's change it in the next PR and merge this.